### PR TITLE
Renew entire action

### LINKED ISSUE #6: Transition to Javascript API
ref #5 

Apparently using the JS API makes this much faster and has more capabilities in general.

### LINKED ISSUE #15: Recommended workflow packaged as action? (deprecation unclear)
Hi @domluna, thanks for providing JuliaFormatter & this formatting workflow! HowI'm wondering what's the reason behind marking the example in the README as deprecated? The linked https://github.com/julia-actions/julia-format/blob/master/workflows/format_check.yml seems rather more messy to add to one's own GitHub Action workflows: are you planning to provide a "packaged" version again at all? Coming from the JuliaFormatter repo, it's quite confusing that there's just this terse deprecation note (and on first reading I did not even realize it applied to the example below). It might be helpful if JuliaFormatter's README linked directly to the format_check.yml (or perhaps even included it in its own repository as an example, instead of depending on a separate repo)?

### PR DESCRIPTION
**Pull request by hyrodium**
fix #6, fix #15.

# Introduction

There are three ways to configure GitHub Actions workflows with [JuliaFormatter.jl](https://github.com/domluna/JuliaFormatter.jl) in this repository.

- Format check with version
    - [format_check.yml](https://github.com/julia-actions/julia-format/blob/master/workflows/format_check.yml)
    - 2019-12-16: Introduced in [this commit](https://github.com/julia-actions/julia-format/commit/fd3e150530cb09c82a88422490598c3f8d58a576)
- Open format PR on interval
    - [format_pr.yml](https://github.com/julia-actions/julia-format/blob/master/workflows/format_pr.yml)
    - 2019-12-16: Introduced in [this commit](https://github.com/julia-actions/julia-format/commit/fd3e150530cb09c82a88422490598c3f8d58a576).
- Use `julia-format` action
    - 2019-09-08: `julia-format` action was introduced in [this commit](https://github.com/julia-actions/julia-format/commit/17a8c4b79634e21010b33b0898f340fe60a6d840).
    - 2019-12-18: The action was deprecated in [this commit](https://github.com/julia-actions/julia-format/commit/fd2eec0b2e91cf24b256fa542eac37426714ba49).
    - 2021-01-12: The deprecation message and usage was removed in [this commit](https://github.com/julia-actions/julia-format/commit/9ea32e8d9b7c16317541d9c6bc7d2901959c6616).
    - 26 repositories still use this action. (https://github.com/julia-actions/julia-format/network/dependents)

This repository is maintained under the [julia-actions](https://github.com/julia-actions) organization, so it would be nice to have well-maintained `julia-format` action in this repository.

# Changes in this PR

- Use [composite action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) instead of [JavaScript action](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action).
    - Note that the composite action was introduced around 2020-08 (https://github.com/actions/runner/issues/438) which is after the issue #6 was created.
- Enable reviewdog for review comments
    - Please check https://github.com/hyrodium/ElasticSurfaceEmbedding.jl/pull/131 for review comments by the bot.
- Update README
- Remove old files

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,57 @@
...
+
+```yaml
+name: Format suggestions
+on:
+  pull_request:
+jobs:
+  code-style:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: julia-actions/julia-format@v2
```
Comment (hyrodium) [inline README.md:15]: I'm thinking of releasing `julia-format` action v2 after merging this PR.

**Comment by hyrodium** [comment]
@domluna Could you please release v2 of this action?

**Comment by domluna** [comment]
ugh I completely forgot how to do that lol. I'll try to figure it out.

**Comment by hyrodium** [comment]
It seems you just need to release v2.0.0 from the following link.
https://github.com/julia-actions/julia-format/releases/new?marketplace=true

Official documentation:

>Create a semantically tagged release. You may also publish to GitHub Marketplace with a simple checkbox. For more information, see "[Managing releases in a repository](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release)" and "[Publishing actions in GitHub Marketplace](https://docs.github.com/en/actions/creating-actions/publishing-actions-in-github-marketplace#publishing-an-action)".

https://docs.github.com/en/actions/creating-actions/releasing-and-maintaining-actions

**Comment by hyrodium** [comment]
@domluna gentle bump

**Comment by hyrodium** [comment]
@domluna 
Thank you for the release!

However, the version tag should be `v2.0.0`, but not [`v.2.0.0`](https://github.com/julia-actions/julia-format/tree/v.2.0.0) :sweat_smile:.
This causes the following error.

>Error: Unable to resolve action `julia-actions/julia-format@v2`, unable to find version `v2`

https://github.com/hyrodium/ElasticSurfaceEmbedding.jl/actions/runs/7013762941/job/19080415961?pr=131

**Comment by hyrodium** [comment]
Oh, I didn't know that we also need manually updating tags such as `v2`, `v2.0`. For example, [julia-docdeploy](https://github.com/julia-actions/julia-docdeploy) has the following tagging.

```bash
commit 580a86b97cef20258d96b53292326e6258dd6859 (HEAD -> master, origin/master, origin/HEAD)
Author: David Widmann <devmotion@users.noreply.github.com>
Date:   Tue Jun 27 19:10:39 2023 +0200

    Use actions/checkout@v3 in README (#30)

commit e62cc8fd639797a0c2922a437d5b1b81c4a12787 (tag: v1.3.1, tag: v1.3, tag: v1, tag: latest)
Author: Hannah Robertson <hannahilea@users.noreply.github.com>
Date:   Wed Mar 29 15:57:14 2023 -0400

    Change check for presence of var (#29)

commit 80c0bdcd0ea677beb566454f5290c4f803b59b64
Author: Rik Huijzer <t.h.huijzer@rug.nl>
Date:   Thu May 26 11:29:48 2022 +0200

    Simplify the key setup instructions (#21)
    
    * Simplify the key setup instructions
    
    * Set `permissions: contents: write`

commit 7fe24b89212e23fda4c760589d7627fb6952b4f0
Author: Rik Huijzer <t.h.huijzer@rug.nl>
Date:   Sun May 22 15:18:53 2022 +0200

    Mention how to set terminal width (#19)

commit 64236d5d0c16adbbd6db9ac6de1f0f728651bdf1
Author: Rik Huijzer <t.h.huijzer@rug.nl>
Date:   Sun May 22 14:40:52 2022 +0200

    Pin example uses to v1 (#20)

commit 9c7706b8dc055756275c9535f2c6c2aed144fc46 (tag: v1.3.0)
Author: Eric Hanson <5846501+ericphanson@users.noreply.github.com>
Date:   Wed Apr 6 15:09:45 2022 +0200

    Make `Pkg.develop` step optional (#18)

commit e6d9f441f8569d719168a1bffbff5cae2a92192b (tag: v1.2.0, tag: v1.2)
Author: Eric Hanson <5846501+ericphanson@users.noreply.github.com>
:
```

>We recommend creating releases using semantically versioned tags – for example, `v1.1.3` – and keeping major (`v1`) and minor (`v1.1`) tags current to the latest appropriate commit. For more information, see "[About custom actions](https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-release-management-for-actions)" and "[About semantic versioning](https://docs.npmjs.com/about-semantic-versioning).

https://docs.github.com/en/actions/creating-actions/releasing-and-maintaining-actions

@domluna
Could you please push tags `v2` and `v2.0`?

**Comment by hyrodium** [comment]
I think it would be nice to install some GitHub action to push tags like `vX` and `vX.Y` from a release `vX.Y.Z`. I couldn't find an appropriate action for that, so let's push tag `v2` and `v2.0` manually for now.

**Comment by domluna** [comment]
I tagged v2.0.0 after the typo-ed version v.2.0.0. Is it still not working?

**Comment by hyrodium** [comment]
>Is it still not working?

Yeah, it still throwing an error like this:

>Error: Unable to resolve action `julia-actions/julia-format@v2`, unable to find version `v2`

https://github.com/hyrodium/ElasticSurfaceEmbedding.jl/actions/runs/7013762941/job/19082358288?pr=131#step:1:35

It seems that we need to push tags such as `v2` and `v2.0` to support the following version specification.

```yaml
      - uses: julia-actions/julia-format@v2.0.0  # works!
      - uses: julia-actions/julia-format@v2      # This is how we recommend in README. This does not work for now.
      - uses: julia-actions/julia-format@v2.0    # Some other people may want to specify a version like this.
```

It would be better if GitHub Actions automatically detects the latest version `v2.0.0` from `v2`, but it seems it does not work like that.

[setup-julia](https://github.com/julia-actions/setup-julia/tags) and [julia-docdeploy](https://github.com/julia-actions/julia-docdeploy/tags) have tags like `v1`, `v1.3`, `v1.3.1`, and `latest`.

**Pull request merged**